client/settings/payment-settings/promotional-banner/re-connect-account-banner.js

```diff
@@ -14,32 +15,57 @@ import {
 	NewPill,
 } from 'wcstripe/settings/payment-settings/promotional-banner/banner-layout';
 
-export const ReConnectAccountBanner = ( { testOauthUrl, oauthUrl } ) => {
+export const ReConnectAccountBanner = () => {
 	const [ isTestModeEnabled ] = useTestMode();
 	const { createErrorNotice } = useDispatch( 'core/notices' );
+	const [ isLoading, setIsLoading ] = useState( false );
 
-	const handleButtonClick = () => {
-		if ( isTestModeEnabled && testOauthUrl ) {
-			recordEvent( 'wcstripe_create_or_connect_test_account_click', {} );
-			recordEvent( 'wcstripe_reconnect_button_click', {
-				source: 're-connect-account-banner',
-				mode: 'test',
-			} );
-			window.location.assign( testOauthUrl );
-		} else if ( ! isTestModeEnabled && oauthUrl ) {
-			recordEvent( 'wcstripe_create_or_connect_account_click', {} );
-			recordEvent( 'wcstripe_reconnect_button_click', {
-				source: 're-connect-account-banner',
-				mode: 'live',
+	const handleButtonClick = async () => {
+		const mode = isTestModeEnabled ? 'test' : 'live';
+
+		recordEvent(
+			isTestModeEnabled
+				? 'wcstripe_create_or_connect_test_account_click'
+				: 'wcstripe_create_or_connect_account_click',
+			{}
+		);
+		recordEvent( 'wcstripe_reconnect_button_click', {
+			source: 're-connect-account-banner',
+			mode,
+		} );
+
+		setIsLoading( true );
+
+		try {
```

Comment:
I just wanted to mention here what I said during our team call today: one way to avoid timeout issues is to add some retry logic. Move the logic inside the block to a helper function and call it a couple of times until we get a valid response. It won't look pretty, but might solve the issue